src/main/scala/com/memsql/streamliner/examples/FileExtractor.scala

```diff
@@ -0,0 +1,58 @@
+package com.memsql.streamliner.examples
+
+import com.memsql.spark.etl.api._
+import com.memsql.spark.etl.utils.PhaseLogger
+import org.apache.spark._
+import org.apache.spark.rdd._
+
+
+// TODO:
+//  1) Figure out how to keep track of file state. we can keep track of names -> md5s in a bucket and periodically check whether any files
+//      have changed.
+//  2) We need to hook into a lower level API in spark to figure out how to expand a file path to an actual list of files (for the above map)
+//  3) We need to persist the map somewhere
+//  4) Support non-'\n' line terminators
+class FileExtractor extends SimpleByteArrayExtractor {
+  var first = true
+  var repeat = false
+  var filePath = ""
+
+  override def initialize(sparkContext: SparkContext, config: UserExtractConfig, batchInterval: Long, logger: PhaseLogger): Unit = {
+    filePath = config.getConfigString("path") match {
+      case Some(s) => s
+      case None => throw new IllegalArgumentException("Missing required argument 'path'")
+    }
+
+    repeat = config.getConfigBoolean("repeat").getOrElse(false)
+
+    if (filePath.startsWith("s3n://")) {
+      val access_key = config.getConfigString("aws_access_key") match {
+        case Some(s) => s
+        case None => throw new IllegalArgumentException("Missing required argument 'aws_access_key' (because path starts with s3n)")
+      }
+      val secret_key = config.getConfigString("aws_secret_key") match {
+        case Some(s) => s
+        case None => throw new IllegalArgumentException("Missing required argument 'aws_secret_key' (because path starts with s3n)")
+      }
+
+      sparkContext.hadoopConfiguration.set("fs.s3n.awsAccessKeyId", access_key)
```

Comment:
oh - maybe something you could do is check if its set first and raise and error if it is.  or you could require that the user sets it in their spark config instead of in the extract config